docs: first iteration on the documentation of the global feature (#2169)
prefix-dev/pixi: docs/features/global_tools.md

### Channels
The channels are the conda channels that will be used to search for the packages.
There is a priority to these, so the first one will have the highest priority, if a package is not found in that channel the next one will be used.
More information on channels can be found [here](../advanced/channel_priority.md).

### Exposed
The exposed binaries are the ones that will be available in the system `PATH`.
This is useful when the package has multiple binaries, but you want to get a select few, or you want to expose it with a different name.
For example, the `python` package has multiple binaries, but you only want to expose the interpreter.
You can do this with the following entry:
```toml
[envs.python]
channels = ["conda-forge"]
dependencies = { python = ">=3.10,<3.11" }
exposed = { python310 = "python" }
```
This also helps when you want to access binaries of dependencies of the package.
For example, the `ansible` package doesn't contain the `ansible` binary, but its dependency `ansible-core`does.
So you can still expose it with:
```
pixi global expose add --environment ansible ansible=ansible

> ```

Reviewer: Considering we plan to auto-expose binaries that have the same name as the package, maybe we choose a different example here?

Author: Changed that

!!! note
    The design for global tools is still in progress, and the commands and behavior may change in future releases.
    The proposal for the global tools feature can be found [here](../design_proposals/pixi_global_manifest.md).

## The Global Manifest
Since `v0.33.0` pixi has a new manifest file that will be created in the global directory (default: `$HOME/.pixi/manifests/pixi-global.toml`).
This file will contain the list of environments that are installed globally, their dependencies and exposed binaries.
The manifest can be edited, synced, checked in to a version control system, and shared with others.

A simple version looks like this:
```toml
[envs.vim]
channels = ["conda-forge"]
dependencies = { vim = "*" } # (1)!
exposed = { vimdiff = "vimdiff", vim = "vim" } # (2)!

[envs.gh]
channels = ["conda-forge"]
dependencies = { gh = "*" }
exposed = { gh = "gh" }

[envs.python]
channels = ["conda-forge"]
dependencies = { python = ">=3.10,<3.11" }
exposed = { python310 = "python" } # (3)!
```

1. Dependencies are the packages that will be installed in the environment. You can specify the version or use a wildcard.

> 2. The exposed binaries are the ones that will be available in the system path. `vim` has multiple so they will all be exposed.

Reviewer: ```suggestion
2. The exposed binaries are the ones that will be available in the system path. `vim` has multiple and all of them are exposed.
```

3. Here python is exposed as `python310` to avoid conflicts with other python installations. You can give it any name you want.

### Channels
The channels are the conda channels that will be used to search for the packages.
There is a priority to these, so the first one will have the highest priority, if a package is not found in that channel the next one will be used.
For example, running:
```
pixi global install --channel conda-forge --channel bioconda snakemake
```
Results in the following entry in the manifest:
```toml
[envs.snakemake]
channels = ["conda-forge", "bioconda"]
dependencies = { snakemake = "*" }
exposed = { snakemake = "snakemake" }
```

More information on channels can be found [here](../advanced/channel_priority.md).

### Exposed
The exposed binaries are the ones that will be available in the system `PATH`.
This is useful when the package has multiple binaries, but you want to get a select few, or you want to expose it with a different name.

> For example, the `python` package has multiple binaries, but you only want to expose the interpreter.

Reviewer: ```suggestion
For example, the `python` package has multiple binaries, but you only want to expose the interpreter as `py3`.
```

Running:
```
pixi global expose add --environment python py3=python3
```
will create the following entry in the manifest:
```toml
[envs.python]
channels = ["conda-forge"]
dependencies = { python = ">=3.10,<3.11" }
exposed = { py3 = "python3" }

> ```

Reviewer: Explain that you can run python by executing `py3`

Author: yes

There is some added automatic behavior, if you install a package with the same name as the environment, it will be exposed with the same name.
Even if the binary name is only exposed through dependencies of the package
For example, running:
```
pixi global install ansible
```
will create the following entry in the manifest:
```toml

> 

Reviewer: ```suggestion
```

### Dependencies
Dependencies are the **Conda** packages that will be installed into your environment. For example, running:
```
pixi global install "python<3.12"
```
creates the following entry in the manifest:
```toml
[envs.vim]
channels = ["conda-forge"]
dependencies = { python = "<3.12" }
# ...
```
Typically, you'd specify just the tool you're installing, but you can add more packages if needed.
Defining the environment to install into will allow you to add multiple dependencies at once.
For example, running:
```shell
pixi global install --environment my-env git vim python
```
will create the following entry in the manifest:
```toml
[envs.my-env]
channels = ["conda-forge"]
dependencies = { git = "*", vim = "*", python = "*" }
# ...
```

You can `add` a dependency to an existing environment by running:
```shell
pixi global install --environment my-env package-a package-b
```
This will be added as dependencies to the `my-env` environment but won't auto expose the binaries from the new packages.

You can `remove` dependencies by running:
```shell
pixi global remove --environment my-env package-a package-b
```

> ### Example: Creating a Data Science Sandbox Environment

Reviewer: This example should come after `Adding a series of tools at once`. It is more complicated

Author: Done

You can create an environment with multiple tools using the following command:
```shell

> pixi global install --environment data-science --expose jupyter=jupyter --expose ipython=ipython jupyter numpy pandas matplotlib ipython

Reviewer: Use fewer packages to avoid that the reader has to scroll

Author: Done
